### PR TITLE
feat: support app-level branding

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

support app-level logos and brand colors for sign-in experience, resolve LOG-9595

- a new search parameter `app_id=<curren-app-id>` will be passed to the sign-in experience
- cache `isDevelopmentTenant` (more to come in the following pulls)
- allow application-level sign-in experience for all apps
- `/api/.well-known/sign-in-exp` now accepts both `organizationId` and `appId` as search parameters
  - organization branding > application branding > default branding
- demo app now accepts other app IDs (via dev panel or search parameter)
- remove sign-in experience prefetch as it needs parameters now (should use ssr for these data in the future)

## Testing

image

integration tests added

## Checklist

- [x] `.changeset`
- [ ] unit tests
- [x] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

File: packages/console/src/pages/ApplicationDetails/ApplicationDetailsContent/Branding/index.tsx [hunk 1]
```diff
@@ -149,32 +165,76 @@ function Branding({ application, isActive }: Props) {
...
+                  {...register('privacyPolicyUrl', {
+                    validate: (value) =>
+                      !value || uriValidator(value) || t('errors.invalid_uri_format'),
+                  })}
+                  error={errors.privacyPolicyUrl?.message}
+                  placeholder="https://"
+                />
+              </FormField>
+            </FormCard>
+          )}
```
Comment (charIeszhao) [inline packages/console/src/pages/ApplicationDetails/ApplicationDetailsContent/Branding/index.tsx:237]: Am I missing something? It seems only third-party apps can update terms of use and privacy policy URLs now?

File: packages/core/src/caches/well-known.ts [hunk 2]
```diff
@@ -13,6 +13,8 @@ type WellKnownMap = {
   'custom-phrases': Record<string, unknown>;
   'custom-phrases-tags': string[];
   'tenant-cache-expires-at': number;
+  // Currently, tenant type cannot be updated once created. So it's safe to cache.
+  'is-developer-tenant': boolean;
```
Comment (charIeszhao) [inline packages/core/src/caches/well-known.ts:17]: nit: `is-development-tenant`?

File: packages/console/src/pages/ApplicationDetails/ApplicationDetailsContent/Branding/index.tsx (hunk 1 above)
Comment (gao-sun) [reply packages/console/src/pages/ApplicationDetails/ApplicationDetailsContent/Branding/index.tsx:237]: yes, it's by design at the moment

File: packages/core/src/caches/well-known.ts (hunk 2 above)
Comment (gao-sun) [reply packages/core/src/caches/well-known.ts:17]: right

File: packages/core/src/oidc/utils.ts [hunk 3]
```diff
@@ -90,6 +90,10 @@ export const buildLoginPromptUrl = (params: ExtraParamsObject, appId?: unknown):
   const searchParams = new URLSearchParams();
   const getSearchParamString = () => (searchParams.size > 0 ? `?${searchParams.toString()}` : '');
 
+  if (appId) {
+    searchParams.append('app_id', String(appId));
+  }
```
Comment (charIeszhao) [inline packages/core/src/oidc/utils.ts:95]: Should we also pass the organization id to the experience page if available?

**Review by charIeszhao — APPROVED** [review]

File: packages/console/src/pages/ApplicationDetails/ApplicationDetailsContent/Branding/index.tsx (hunk 1 above)
Comment (charIeszhao) [reply packages/console/src/pages/ApplicationDetails/ApplicationDetailsContent/Branding/index.tsx:237]: ok

File: packages/core/src/libraries/sign-in-experience/index.ts [hunk 4]
```diff
@@ -196,7 +224,10 @@ export const createSignInExperienceLibrary = (
     };
 
     return {
-      ...deepmerge(signInExperience, organizationOverride ?? {}),
+      ...deepmerge(
+        deepmerge(signInExperience, appSignInExperience ?? {}),
```
Comment (simeng-li) [inline packages/core/src/libraries/sign-in-experience/index.ts:228]: not a blocker, should not merge application-level SIE with default SIE for the third-party apps.

**Review by simeng-li — APPROVED** [review]
LGTM

File: packages/core/src/libraries/sign-in-experience/index.ts (hunk 4 above)
Comment (gao-sun) [reply packages/core/src/libraries/sign-in-experience/index.ts:228]: i'll update it with some tests in a new pull

**Pull request merged**